.github/workflows/test-library.yml

```diff
@@ -709,15 +716,89 @@ jobs:
         docker buildx use proxypybuilder
         docker buildx inspect
         docker buildx ls
-    - name: Set PROXYPY_CONTAINER_VERSION
-      run: |
-        echo "PROXYPY_CONTAINER_VERSION=$(echo '${{ needs.pre-setup.outputs.dist-version }}' | tr + .)" > $GITHUB_ENV
     - name: Build container
-      run: |
-        make container-buildx \
-          -e PROXYPY_PKG_PATH='dist/${{ needs.pre-setup.outputs.wheel-artifact-name }}' \
-          -e BUILDX_TARGET_PLATFORM='${{ matrix.targetplatform }}' \
-          -e PROXYPY_CONTAINER_VERSION='${{ env.PROXYPY_CONTAINER_VERSION }}'
+      run: >-
+        docker images;
+        make container-buildx
+        -e PROXYPY_PKG_PATH='dist/${{
+          needs.pre-setup.outputs.wheel-artifact-name
+        }}'
+        -e BUILDX_TARGET_PLATFORM='${{ matrix.targetplatform }}'
+        -e PROXYPY_CONTAINER_VERSION='${{
+          needs.pre-setup.outputs.container-version
+        }}';
+        docker images
+    - name: Run & Test container
+      run: >-
+        docker run
+        -d
+        -p 8899:8899
+        abhinavsingh/proxy.py:${{
+          needs.pre-setup.outputs.container-version
+        }}
+        --hostname 0.0.0.0
+        --enable-web-server
+        --local-executor && ./tests/integration/test_integration.sh 8899
+    - name: Login to GHCR
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
This context is only available for the event triggers that are safe and don't have a high risk of running untrusted code contributed by random people via PRs.

Reply:
+1 , also updated repo settings to only allow whitelisted actions.